src/components/GroupForm/BulkUserManager/hooks/useMap.js

```diff
@@ -0,0 +1,29 @@
+import { useState, useMemo } from 'react'
+
+export const useMap = () => {
+    const [map, setMap] = useState(new Map())
+
+    return useMemo(
```

Comment:
`useMemo` ensures a that the reference to the 'map' returned by `useMap` is stable unless mutated, allowing the following pattern:

```js
const someMap = useMap()

useEffect(() => {
  // someMap has been mutated, so trigger e.g. onChange handlers
}, [someMap])
```